### PR TITLE
feat: Adds delete option for messages in conversation

### PR DESCRIPTION
**Pull request by nithindavid**
image

### CODE REVIEW

**Review by iamsivin — DISMISSED** [review]

File: app/javascript/dashboard/modules/conversations/components/MessageContextMenu.vue [hunk 1]
```diff
@@ -0,0 +1,75 @@
+<template>
+  <div class="context-menu">
+    <woot-button
+      icon="ion-more"
```
Comment (pranavrajs) [inline app/javascript/dashboard/modules/conversations/components/MessageContextMenu.vue:4]: Can we use 20px for this icon? It seems really small.

File: app/javascript/dashboard/components/widgets/conversation/Message.vue [hunk 2]
```diff
@@ -211,6 +234,29 @@ export default {
...
     },
   },
+  methods: {
+    handleContextMenuClick() {
+      this.showContextMenu = !this.showContextMenu;
+    },
+    async handleDelete() {
+      const { conversation_id: conversationId, id: messageId } = this.data;
+      try {
+        await this.$store.dispatch('deleteMessage', {
```
Comment (pranavrajs) [inline app/javascript/dashboard/components/widgets/conversation/Message.vue:244]: [vuex] unknown action type: deleteMessage

File: app/javascript/dashboard/i18n/locale/en/conversation.json [hunk 3]
```diff
@@ -27,6 +27,8 @@
     "REMOVE_SELECTION": "Remove Selection",
     "DOWNLOAD": "Download",
     "UPLOADING_ATTACHMENTS": "Uploading attachments...",
+    "SUCCESS_DELETE_MESSSAGE": "Message deleted successfully",
```
Comment (pranavrajs) [inline app/javascript/dashboard/i18n/locale/en/conversation.json:30]: Please correct the spelling.

**Review by pranavrajs — CHANGES_REQUESTED** [review]
I could get the option to work. When I click on delete message, it throws an error.

File: app/javascript/dashboard/components/widgets/conversation/Message.vue (hunk 2 above)
Comment (pranavrajs) [reply app/javascript/dashboard/components/widgets/conversation/Message.vue:244]: Never mind, I merged the store, it worked.

**Comment by pranavrajs** [comment]
On QA, I figured that the dropdown goes behind the side panel.

Screenshot 2021-06-29 at 11 20 19 AM

**Comment by nithindavid** [comment]
> On QA, I figured that the dropdown goes behind the side panel.
> 
> Screenshot 2021-06-29 at 11 20 19 AM

@pranavrajs Our current menu doesn't change position by context. Might need another PR to fix this.

**Comment by pranavrajs** [comment]
Can't we add `position: fixed` to the dropdown pane so that it would show something like the one below? That way we don't need to put effort to fix the positioning based on context.

Screenshot 2021-06-29 at 2 05 05 PM

Screenshot 2021-06-29 at 2 04 59 PM

**Comment by nithindavid** [comment]
@pranavrajs It won't move on scroll if we put fixed

**Comment by nithindavid** [comment]
@pranavrajs here's a solution without fixed, Making the menu position by message type (left or right)

image

**Comment by pranavrajs** [comment]
@nithindavid Works like a charm! Nice work!

**Pull request merged**